DiscordChatExporter.Core/Exporting/Filtering/BooleanMessageFilter.cs

```diff
@@ -0,0 +1,27 @@
+﻿using DiscordChatExporter.Core.Discord.Data;
+
+namespace DiscordChatExporter.Core.Exporting.Filtering
+{
+    public class BooleanMessageFilter : MessageFilter
+    {
+        private readonly MessageFilter _first;
+        private readonly MessageFilter _second;
+        private readonly bool _unioned;
```

Comment:
I'd probably use an enum field `_kind` for clarity:

```
public enum BinaryExpressionKind
{
    Or,
    And
}
```

To me, "unioned" doesn't really convey the right meaning.